src/Schema/Comparator.php

```diff
@@ -44,6 +48,32 @@ public function __construct(?AbstractPlatform $platform = null)
         $this->platform = $platform;
     }
 
+    /**
+     * @param array{Schema, Schema} $args
+     */
+    public function __call(string $method, array $args): SchemaDiff
+    {
+        if ($method !== 'compareSchemas') {
+            throw new BadMethodCallException(sprintf('Unknown method "%s"', $method));
+        }
+
+        return $this->doCompareSchemas(...$args);
+    }
+
+    /**
+     * @param array{Schema, Schema} $args
+     */
+    public static function __callStatic(string $method, array $args): SchemaDiff
+    {
+        if ($method !== 'compareSchemas') {
+            throw new BadMethodCallException(sprintf('Unknown method "%s"', $method));
+        }
+
```

Comment:
I think, we can emit the deprecation here. The static call is actually what we're deprecating.

Reply:
@derrabus you're right, but let's do that on 3.4.x after merging up?